### PR TITLE
Fares backup

### PR DESCRIPTION
**Pull request by orzeuek**

### CODE REVIEW

File: src/backup/Storage.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface Storage {
+  persist(filePath: string): Promise<void>;
```
Comment (yezooz) [inline src/backup/Storage.ts:2]: Not sure if throwing an exception is a good pattern, especially in this case. Wouldn't it be better to return some error type, or at least a boolean?

File: src/cli/BackupDatabaseCommand.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  protected async doBackup(): Promise<string> {
+    const backupFileName = "/tmp/" + this.databaseName + "_dump_" + new Date().toISOString() + ".sql";
+    const command = "mysqldump -u " + this.username +
+      " --password=" + this.password +
+      " -h " + this.host +
+      " " + this.databaseName +
+      " > " + backupFileName;
+    await this.cmd(command);
```
Comment (yezooz) [inline src/cli/BackupDatabaseCommand.ts:46]: How do we know if creds failed, network timeout or some other random error? Does it throw an exception?

Comment (orzeuek) [reply src/cli/BackupDatabaseCommand.ts:46]: yup, if exit code of process is != 0 then it's propagated as exception.

File: src/backup/Storage.ts (hunk 1 above)
Comment (orzeuek) [reply src/backup/Storage.ts:2]: I disagree. 
You want to persist a file. It's a commend. You don't expect any output from command. 
Command can't be executed... It's unexpected, so it's exception. 
Client (method client) know that commands do not return any value, but can fail. If client want to handle that, he should use try/catch. If he don't know how to handle that, just propagate exception.

**Review by yezooz — APPROVED** [review]

**Pull request merged**